src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,76 @@
+package duke;
+
+import duke.exception.EmptyDescriptionException;
+import duke.exception.MissingParameterException;
+import duke.exception.WrongCommandException;
+
+public class Parser {
+
+    public static String parseInput(String command) {
+        return command.toLowerCase().trim();
+    }
+
+    public static int parseNumber(String command) throws EmptyDescriptionException, NumberFormatException {
+        if (command.split(" ").length < 2) {
+            throw new EmptyDescriptionException();
+        }
+        return Integer.parseInt(command.split(" ")[1]);
+    }
+
+
+    public static String[] parseTask(String command, Action taskType) throws EmptyDescriptionException, MissingParameterException {
+        String[] parameters = new String[2];
+        //Check if the description for is empty
+        if (parseInput(command).split(" ").length < 2) {
+            throw new EmptyDescriptionException();
+        }
+        switch (taskType) {
+        case TO_DO:
+            parameters[0] = command.substring(command.indexOf("todo") + 5);
+            parameters[1] = "N/A";
+            break;
+        case EVENT:
+            if (command.indexOf("/at") == -1) {
+                throw new MissingParameterException();
+            }
+            parameters[0] = command.substring(command.indexOf("event") + 6, command.indexOf("/at"));
+            parameters[1] = command.substring(command.indexOf("/at") + 3);
+            break;
+        case DEADLINE:
+            if (command.indexOf("/by") == -1) {
+                throw new MissingParameterException();
+            }
+            parameters[0] = command.substring(command.indexOf("deadline") + 9, command.indexOf("/by"));
```

Comment:
Note that single line of code [should not exceed >120 characters](https://se-education.org/guides/conventions/java/basic.html#layout). Consider formatting complex strings like this in multiple lines.